### DISCUSSION TITLE
Dedicated model for ASR

### DISCUSSION THREAD

**Discussion by gupta9ankit5** [facebookresearch/seamless_communication#273] [Q&A] [answered]
Hello, I recently came across this repository and am currently in the exploration phase. While Automatic Speech Recognition (ASR) is mentioned throughout the documentation as one of the model's utilities, alongside others such as S2ST, S2TT, T2ST, and T2TT, I couldn't locate a dedicated model specifically for ASR. Could someone please guide me on how to use the model for ASR?

**Comment by elbayadm** [comment, marked as answer]
Hi, @gupta9ankit5 !
This is an example script to run ASR only:
I'm testing with this audio:
```bash
wget https://dl.fbaipublicfiles.com/seamlessM4T/LJ037-0171_sr16k.wav -O example.wav
```

```python
import torch
from seamless_communication.inference import Translator
from seamless_communication.inference.generator import SequenceGeneratorOptions

device = torch.device("cuda:0")
dtype = torch.float16
model_name = "seamlessM4T_v2_large"
audio = "example.wav"
decoding_options = SequenceGeneratorOptions(beam_size=5, soft_max_seq_len=(1, 200))
tgt_lang = "eng"

translator = Translator(
    model_name,
    vocoder_name_or_card=None,
    device=device,
    dtype=dtype,
    input_modality='speech',
    output_modality='text'
)
# If you want to look at the model architecture (although it's unified, we can load only the speech encoder and text decoder for ASR/S2TT)
# print('Model arch:', translator.model)

transcript, _ = translator.predict(
    input=audio,
    task_str="ASR",
    tgt_lang=tgt_lang,
    text_generation_opts=decoding_options,
)

print('transcript:', transcript[0])
```
Output:
```
transcript: the examination and testimony of the experts enabled the commission to conclude that five shots may have been fired
```
You can also use the `m4t_predict` command with:
```bash
m4t_predict "example.wav" --task asr --tgt_lang eng --model_name seamlessM4T_v2_large
```
```
output:
2023-12-14 21:39:48,342 INFO -- seamless_communication.cli.m4t.predict.predict: Translated text in eng: the examination and testimony of the experts enabled the commission to conclude that five shots may have been fired
```

Another alternative is to use the [dedicated models from Hugging Face](https://huggingface.co/docs/transformers/main/en/model_doc/seamless_m4t_v2#1-use-dedicated-models)

**Reply by gupta9ankit5** [reply]
Please, let me know if I understood it right or not. 
So for ASR, basically you are using the model for S2TT while keeping both source and target languages the same.

**Reply by elbayadm** [reply]
That's right, we treat ASR as a special case of S2TT where `src_lang = tgt_lang`.

**Reply by RRThivyan** [reply]
Hi, I have come across this post about ASR. Is there any way to identify the source language of the input audio also? I am working in a project to get speech input and I have to identify the language too for further process. Is this model allows us to detect input source language too?
